File: Introduction to Robotics Course/Chapter1/Lesson2/Chapter1_Lesson2.cpp

#include 
#include 

std::string classifySystem(double cT, double iEU, int rR,
                           double th_c=0.3, double th_i=0.2, int th_r=2){
    if(cT < th_c && iEU < th_i && rR == 1) return "Automation";
    if(cT >= th_c && iEU >= th_i && rR >= th_r) return "Robot";
    return "Mechatronic System";
}

int main(){
    std::cout << "CNC line -> " << classifySystem(0.1, 0.05, 1) << "\n";
    std::cout << "Active suspension -> " << classifySystem(0.4, 0.15, 2) << "\n";
    std::cout << "Mobile robot -> " << classifySystem(0.7, 0.6, 5) << "\n";
    return 0;
}
      

File: Introduction to Robotics Course/Chapter1/Lesson2/Chapter1_Lesson2.py

import numpy as np

def classify_system(c_T, i_EU, r_R,
                    th_c=0.3, th_i=0.2, th_r=2):
    """
    c_T: task capacity proxy in [0,1]
    i_EU: normalized environment-action coupling proxy in [0,1]
    r_R: reprogrammability rank proxy (integer >=1)
    """
    if c_T < th_c and i_EU < th_i and r_R == 1:
        return "Automation"
    if c_T >= th_c and i_EU >= th_i and r_R >= th_r:
        return "Robot"
    return "Mechatronic System"

examples = [
    ("CNC line",        0.1, 0.05, 1),
    ("Active suspension",0.4, 0.15, 2),
    ("Mobile robot",    0.7, 0.6, 5),
]

for name, cT, iEU, rR in examples:
    print(name, "->", classify_system(cT, iEU, rR))
      